src/api/route-services/projects.js

```diff
@@ -184,9 +184,11 @@ class ProjectsService {
       const { experiments } = await this.getProject(projectUuid);
 
       if (experiments.length > 0) {
```

Comment:
I saw in the logs that the line `experiments.length` sometimes throws an error, specifically when `experiments` is none -- it complains that there is not such thing as length of undefined. Will be good to add a check in the code to guard against those cases and return not found `experiments` is undefined

Reply:
Yea, it happens when we're loading projects for an experiment. This error always pop up if we switch to experiments without project (e.g. previously created experiments). I think it'll need fixes in the UI and API.

Reply:
What are the implications of this error? Is it just something annoying that shows up in logs all the time? What is its impact to the user?

Reply:
also, is it a small thing to fix? Do you think the fix makes sense to be in a separate ticket?

Reply:
Added checks around the call. This function is also called when checking credentials. There are still errors being shown in the UI, rendering errors for unrelated experiment. I'm not sure why, but it seems like the notification is "cached" or not entirely cleared.

Reply:
Moved into ticket https://biomage.atlassian.net/browse/BIOMAGE-1012

Reply:
thanks, let's look into this there